core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryPresenter.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.history
+
+import android.util.Log
+import io.reactivex.Observable
+import io.reactivex.Scheduler
+import org.kiwix.kiwixmobile.core.base.BasePresenter
+import org.kiwix.kiwixmobile.core.data.DataSource
+import org.kiwix.kiwixmobile.core.di.qualifiers.Computation
+import org.kiwix.kiwixmobile.core.di.qualifiers.MainThread
+import org.kiwix.kiwixmobile.core.history.HistoryContract.Presenter
+import org.kiwix.kiwixmobile.core.history.HistoryContract.View
+import org.kiwix.kiwixmobile.core.history.HistoryListItem.HistoryItem
+import java.util.Locale
+import javax.inject.Inject
+
+internal class HistoryPresenter @Inject constructor(
+  private val dataSource: DataSource,
+  @param:MainThread private val mainThread: Scheduler,
+  @param:Computation private val computation: Scheduler
+) : BasePresenter<View>(), Presenter {
+
+  override fun loadHistory(showHistoryCurrentBook: Boolean) {
+    dataSource.getDateCategorizedHistory(showHistoryCurrentBook)
```

Comment:
I must have introduced some confusion here, you still need the disposable variable, it holds on to the previous invocation of `loadHistory`, it there is a load in progress we want to cancel it and start a new one.

This should be never loading because it immediately disposes itself after calling subscribe